staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -43,67 +47,100 @@ type FeatureSupportChecker interface {
 	// Supports check if the feature is supported or not by checking internal cache.
 	// By default all calls to this function before calling CheckClient returns false.
 	// Returns true if all endpoints in etcd clients are supporting the feature.
-	Supports(feature storage.Feature) (bool, error)
+	Supports(feature storage.Feature) bool
```

Comment:
I'm not sure if you've already discussed this, but should users of this method be able to distinguish between `not supported` and `I don’t yet know because I haven’t been able to check`?

Reply:
We use `progressNotifySupported    *bool` for that reason, but either we should wait for all clients to been checked or return `false`. And waiting for all clients is not applicable here. One other solution might be returning `*bool` from here but i'm not confident enough about it (not sure we can or not as this is a good practice or not), and also needs approval from @wojtek-t and @serathius too.

Reply:
> I'm not sure if you've already discussed this, but should users of this method be able to distinguish between `not supported` and `I don’t yet know because I haven’t been able to check`?

How that distinction would be used by user? In case of consistent reads, we don't need it.

Reply:
Let's assume that I would like to request a bookmark for the purpose of initializing some object. If I could distinguish between the states, then I would know whether to wait for information from the checker or not.

More specifically, such functionality would be useful to fix https://github.com/kubernetes/kubernetes/pull/124754#discussion_r1594063439

Reply:
`purpose of initializing some object.` can you clarify what do you mean by that?

Reply:
We found a bug in the cacher, it won't deliver bookmark events until it receives an event from the underlying storage. We were thinking about scheduling at least one progress notification request so that the cacher is primed. Knowing whether the version/feature checker has been initialised would help in this scenario.

Reply:
Ok, I would need more context to understand the case more. Can you send me a issue number?

Reply:
I think there is relatively non-invasive change to make it work for the StreamingList scenario too. But let's proceeed with unblocking at least consistent lists and we will make it work for streaming lists as a followup.

Reply:
OK, if you don't mind, could you shed some light on the potential solution here?

Reply:
https://github.com/kubernetes/kubernetes/pull/124754/files#r1596607679

Reply:
OK, this seems to solve one case. I wonder what would happen in a situation where a watch request actually requires progress notification. In that case changing the state from `on` to `off` might/will be problematic - the request will wait for the notification which might not come to the server.

Reply:
We should be able to catch that at runtime and fail the request in such case. This should be extremely rare (it literally should never happen) and for that reason I believe it's fine.